src/error.rs

```diff
@@ -1,15 +1,27 @@
 //! Error types.
 
-use thiserror::Error;
+use std::fmt::{Display, Formatter, Result};
```

Comment:
Don't import `std::fmt::Result`. Personally I prefer not to import these types at all but just qualify them everywhere, but `Result` is particularly bad because it overwrites the normal `Result` from the prelude.